rollup/sync_service.go

```diff
@@ -1,641 +1,249 @@
-	GetNumPendingQueueElements(*bind.CallOpts) (*big.Int, error)
-	GetQueueElement(*bind.CallOpts, *big.Int) (ctc.Lib_OVMCodecQueueElement, error)
-	GetTotalElements(*bind.CallOpts) (*big.Int, error)
-}
-
-type ExecutionManagerCaller interface {
-	GetMaxTransactionGasLimit(opts *bind.CallOpts) (*big.Int, error)
-}
-
-type RollupTxsByIndex []*RollupTransaction
-
-func (l RollupTxsByIndex) Len() int           { return len(l) }
-func (l RollupTxsByIndex) Swap(i, j int)      { l[i], l[j] = l[j], l[i] }
-func (l RollupTxsByIndex) Less(i, j int) bool { return l[i].index < l[j].index }
-
-// RollupTransaction represents a transaction parsed from L1
-type RollupTransaction struct {
-	tx          *types.Transaction
-	blockHeight uint64
-	index       uint64
-	executed    bool
-}
-
-// Move this to its own file
-func NewTransactionCache() *TransactionCache {
-	return &TransactionCache{
-		m: new(sync.Map),
-	}
-}
-
-type TransactionCache struct {
-	m *sync.Map
-}
-
-func (t *TransactionCache) Store(index uint64, rtx *RollupTransaction) {
-	t.m.Store(index, rtx)
-}
-
-func (t *TransactionCache) Delete(index uint64) {
-	t.m.Delete(index)
-}
-
-func (t *TransactionCache) Load(index uint64) (*RollupTransaction, bool) {
-	result, ok := t.m.Load(index)
-	if !ok {
-		return nil, false
-	}
-	rtx, ok := result.(*RollupTransaction)
-	if !ok {
-		log.Error("Incorrect type in transaction cache", "type", fmt.Sprintf("%T", rtx))
-		return nil, false
-	}
-	return rtx, true
-}
-
-func (t *TransactionCache) Range(f func(uint64, *RollupTransaction)) {
-	t.m.Range(func(key interface{}, value interface{}) bool {
-		rtx, ok := value.(*RollupTransaction)
-		if !ok {
-			log.Error("Unexpected value in transaction cache", "type", fmt.Sprintf("%T", value))
-			return true
-		}
-		index, ok := key.(uint64)
-		if !ok {
-			log.Error("Unexpected key type in transaction cache", "type", fmt.Sprintf("%T", key))
-			return true
-
-		}
-		f(index, rtx)
-		return true
-	})
-}
-
-// These variables represent the event signatures
-var (
-	transactionEnqueuedEventSignature      = crypto.Keccak256([]byte("TransactionEnqueued(address,address,uint256,bytes,uint256,uint256)"))
-	queueBatchAppendedEventSignature       = crypto.Keccak256([]byte("QueueBatchAppended(uint256,uint256,uint256)"))
-	sequencerBatchAppendedEventSignature   = crypto.Keccak256([]byte("SequencerBatchAppended(uint256,uint256,uint256)"))
-	transactionBatchAppendedEventSignature = crypto.Keccak256([]byte("TransactionBatchAppended(uint256,bytes32,uint256,uint256,bytes)"))
-)
-
-// Eth1Data represents the last processed ethereum 1 data
-// The sync service updates this as it syncs blocks.
-type Eth1Data struct {
-	BlockHeight uint64
-	BlockHash   common.Hash
-}
-
+// TODO: rename this to OVMContext
 // LatestL1ToL2 represents the latest blocknumber and timestamp.
-// This must be separate than Eth1Data because it is only updated
-// each time that there is a L1 to L2 transaction.
 type LatestL1ToL2 struct {
```

Comment:
This really should be renamed to `EthContext` and `queueIndex` and `index` are dead code because they were added to a leveldb index